guac_core/src/channel_client/channel_manager.rs

```diff
@@ -347,31 +351,35 @@ impl ChannelManager {
                 );
 
                 ensure!(
-                    state.my_state().deposit_b == 0.into(),
+                    state.my_state().deposit_b == 0u32.into(),
                     "Their deposit must be 0 to begin with"
                 );
 
                 ensure!(
-                    state.their_state().deposit_b == 0.into(),
+                    state.their_state().deposit_b == 0u32.into(),
                     "Their deposit must be 0 to begin with"
                 );
 
                 *state.my_state_mut().their_deposit_mut() = state
                     .my_state_mut()
                     .their_deposit_mut()
+                    .clone()
                     .add(chan.deposit_b.clone());
                 *state.my_state_mut().their_balance_mut() = state
                     .my_state_mut()
                     .their_balance_mut()
+                    .clone()
                     .add(chan.deposit_b.clone());
 
                 *state.their_state_mut().their_deposit_mut() = state
```

Comment:
isn't this double billing?

Reply:
Good spot. Looks suspicious and it probably slipped in my earlier PRs. Will update.